src/SDK/Trace/Sampler/ParentBased.php

```diff
@@ -29,34 +29,22 @@
  */
 class ParentBased implements SamplerInterface
 {
-    private SamplerInterface $remoteParentSampler;
-
-    private SamplerInterface $remoteParentNotSampler;
-
-    private SamplerInterface $localParentSampler;
-
-    private SamplerInterface $localParentNotSampler;
-
     /**
      * ParentBased sampler delegates the sampling decision based on the parent context.
      *
      * @param SamplerInterface $root Sampler called for the span with no parent (root span).
-     * @param SamplerInterface|null $remoteParentSampler Sampler called for the span with the remote sampled parent. When null, `AlwaysOnSampler` is used.
-     * @param SamplerInterface|null $remoteParentNotSampler Sampler called for the span with the remote not sampled parent. When null, `AlwaysOffSampler` is used.
-     * @param SamplerInterface|null $localParentSampler Sampler called for the span with local the sampled parent. When null, `AlwaysOnSampler` is used.
-     * @param SamplerInterface|null $localParentNotSampler Sampler called for the span with the local not sampled parent. When null, `AlwaysOffSampler` is used.
+     * @param SamplerInterface $remoteParentSampler Sampler called for the span with the remote sampled parent. When null, `AlwaysOnSampler` is used.
+     * @param SamplerInterface $remoteParentNotSampler Sampler called for the span with the remote not sampled parent. When null, `AlwaysOffSampler` is used.
+     * @param SamplerInterface $localParentSampler Sampler called for the span with local the sampled parent. When null, `AlwaysOnSampler` is used.
+     * @param SamplerInterface $localParentNotSampler Sampler called for the span with the local not sampled parent. When null, `AlwaysOffSampler` is used.
```

Comment:
Changing to non-nullable parameters is a breaking change.
If we still want to change it: should remove the `When null, ...` part from doc

Reply:
Reverted to the original code, and disabled a rector rule for this file.